### ISSUE TITLE
Incorrect match in JSX with space-infix-ops rule

### ISSUE THREAD

**Issue by aggieben** [eslint/eslint#3534]
I have a function with embedded JSX like this:
```javascript
    _renderNode(node, i) {
        if (!node) {
            return null;
        } else {
            return (
                (node.children && node.children.length > 0)
                ? (
                    <li key={i}>{node.Name}
                        <ul>
                            {_.map(node.children, (val, idx) => this._renderNode(val, idx))}
                        </ul>
                    </li>
                )
                : (
                    <li key={i}>{node.Name}</li> /* space-infix-ops hits on this line */
                )
            );
        }
    }
```

I suspect that the rule hits becaus of the `}<{` sequence.

version: 2.10.1

eslintrc:
```javascript
{
  "ecmaFeatures": {
    "jsx": true,
    "modules": true
  },
  "env": {
    "browser": true,
    "node": true
  },
  "parser": "babel-eslint",
  "rules": {
    "quotes": [2, "single"],
    "eol-last": [0],
    "no-mixed-requires": [0],
    "no-underscore-dangle": [0],
    "no-unused-vars": [1, {"vars": "all", "args": "none"}],
    "indent": 2,
    "brace-style": [2, "1tbs", { "allowSingleLine": true }],
    "camelcase": [2, {"properties": "always"}],
    "comma-style": [2, "last"],
    "no-multiple-empty-lines": [2, {"max": 1}],
    "no-multi-spaces": [1, {
      "exceptions": {
          "VariableDeclarator": true
      }
    }],
    "new-cap": [2, {
      "capIsNewExceptions": [ "ClassNames" ]
    }],
    "react/display-name": 0,
    "react/jsx-boolean-value": 1,
    "react/jsx-no-undef": 1,
    "react/jsx-quotes": 1,
    "react/jsx-sort-prop-types": 1,
    "react/jsx-sort-props": 0,
    "react/jsx-uses-react": 1,
    "react/jsx-uses-vars": 1,
    "react/no-did-mount-set-state": 1,
    "react/no-did-update-set-state": 1,
    "react/no-multi-comp": 1,
    "react/no-unknown-property": 1,
    "react/prop-types": 0,
    "react/react-in-jsx-scope": 1,
    "react/self-closing-comp": 1,
    "react/sort-comp": 0,
    "react/wrap-multilines": 1
  },
  "plugins": [
    "react"
  ]
}
```

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by ilyavolodin** [comment]
Can you post exact console output of ESLint run please?

**Comment by aggieben** [comment]
```
C:\Users\Benjamin\Projects\Dir1\cm-core-fe [cm-store +4 ~7 -0 !]
λ npm run lint

> church-management-core@1.0.0 lint C:\Users\Benjamin\Projects\Dir1\cm-core-fe
> eslint src


src\js\components\ChurchStructure\ChurchStructureMap.react.js
  13:6   warning  _rotate is defined but never used  no-unused-vars
  76:12  warning  line is defined but never used     no-unused-vars
  89:12  warning  country is defined but never used  no-unused-vars

src\js\components\ChurchStructure\ChurchStructureOutline.react.js
  26:0   error  Multiple blank lines not allowed  no-multiple-empty-lines
  41:20  error  Infix operators must be spaced    space-infix-ops

src\js\stores\User\UserClaimStore.js
  24:9  warning  update is defined but never used  no-unused-vars
```

**Comment by nzakas** [comment]
Can you help us out by indicating the line numbers in your code? It's hard to tell what's going on without being able to match those up.

**Comment by aggieben** [comment]
I left a comment in the first code snippet to point out exactly where the issue is, but maybe that wasn't clear.  Here's a screenshot:

![image](https://cloud.githubusercontent.com/assets/1321/9508725/c1dadb00-4c1d-11e5-824a-ca528dbaa495.png)

**Comment by BYK** [comment]
`"parser": "babel-eslint"` - I'd bet this is a Babel issue.

**Comment by nzakas** [comment]
Can you try using the default parser and see if you still have the same problem?

**Comment by ilyavolodin** [comment]
No response, so closing. Feel free to reopen if this is still an issue.
